### PR TITLE
Rewrite to use only BitBucket v2.0 APIs

### PR DESCRIPTION
**Pull request by d2roth**
Because BitBucket is deprecating API version 1.0 I have rewritten the BitBucket JSON script to work with their new API. I have done preliminary testing but I would like others to review it as well.

I started with a fresh file though did use many of the concepts and principles from the previous BBjson.php file.

For PowerShell here were my two test scripts to create commits:

```BASH
# Multiple commits to one file
For ($i = 1; $i -lt 5; $i++){
  For ($x = 1; $x -lt 15; $x++){
    echo "other contents ${x}" >> "test-${i}.txt";
    git add -A;
    git commit -am "Committing file ${i}";
  }
}
```
and 
```BASH
# Multiple commits to multiple files
For ($i = 1; $i -lt 100; $i++){
  For ($x = 1; $x -lt 5; $x++){
    echo "other contents ${x}-${i}" >> "test-${x}-${i}.txt";
  }
  git add -A;
  git commit -am "Committing file ${i}";
}
```

If you find any issues or have questions I am open to modifying it. I am planning on using this where I work but improvements would be welcomed.

### CODE REVIEW

**Comment by pankydcoder** [comment]
@d2roth  I just saw this pull request. I have raised the same [https://github.com/Wanchai/FTPbucket/issues/25](issue) i guess. Can you please confirm this if it will get resolved with the pull request? If yes, i am eagerly waiting for it to get merged to master.

**Comment by d2roth** [comment]
@pankydcoder, when I made this pull request I had tested everything locally. Since then work for bust so I haven't implement this yet. I did however run into the same issue with API version 1.0 being deprecated so I will try and implement this code in the next couple of days and respond here whether things work properly again.

**Comment by pankydcoder** [comment]
@d2roth Great. Thankyou!!

**Comment by pankydcoder** [comment]
Hi @d2roth , Did you get chance to work on this issue?

**Comment by d2roth** [comment]
Hey @pankydcoder ,
I am looking through this now and I realize I still used some 1.0 endpoints so this pull request will not fix the issue. 

The new problem is that BitBucket no longer supports changesets so I think we would need to change to using diffstat and parse through it to find the files that were changed.

The API documentation for diffstat is here: https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/diffstat/%7Bspec%7D

I will continue to look into this but it is looking fairly complicated to get the list of files that were changed between two commits.

**Comment by d2roth** [comment]
Hey @pankydcoder ,
I have been able to get the list of files with running diffstat. I still have more testing to do and to complete rewriting all of the methods but I think I can get this back up and working with BitBucket 2.0 APIs.

I will update here when I know more.

**Comment by pankydcoder** [comment]
@d2roth Glad to know that you got the way. Thanks a lot for all the efforts you are putting in to get this working.

File: ftpbucket/BBjson.php [hunk 1]
```diff
@@ -1,311 +1,366 @@
...
+    private function load_config() {
+
+        if ( !is_file ( 'config.php' ) ) $this->error ( 'Can\'t find config.php' ); 
+        $config = include 'config.php';
+
+        // Make sure we only get 1 configuration
+        $check = 0;
+
+        // The name passed through for the repository is only the full name so replace the owner part with nothing to get just the repository slug
+        $payload_repo_name = trim( str_replace( $this->payload['repository']['owner']['username'], '', $this->payload['repository']['full_name'] ), '/' );
```
Comment (Wanchai) [inline ftpbucket/BBjson.php]: I don't get this line. What is it for?
When I test it, it crashes when $payload_repo_name is tested against $repo['repo_name'].

I get my-name/my-repo != my-repo

Comment (Wanchai) [reply ftpbucket/BBjson.php]: Ok, on my payload $this->payload['repository']['owner']['username'] does not exist.

I gonna fix this.

**Pull request merged**